src/main/kotlin/uk/gov/dluhc/printapi/service/PrintResponseFileService.kt

```diff
@@ -17,11 +19,28 @@ class PrintResponseFileService(
         val printResponsesString = sftpService.fetchFileFromOutBoundDirectory(directory, fileName)
         val printResponses = parsePrintResponseContent(printResponsesString)
         printResponseProcessingService.processBatchAndPrintResponses(printResponses)
-        sftpService.removeFileFromOutBoundDirectory(directory, fileName)
+        removeFile(directory, fileName)
     }
 
     private fun parsePrintResponseContent(printResponsesString: String): PrintResponses {
         logger.debug { "Parsing print responses $printResponsesString" }
         return objectMapper.readValue(printResponsesString, PrintResponses::class.java)
     }
+
+    private fun removeFile(directory: String, fileName: String) {
+        try {
+            sftpService.removeFileFromOutBoundDirectory(directory, fileName)
+        } catch (ex: MessagingException) {
+            if (isCauseNoSuchFile(ex)) {
+                logger.error("File $fileName was not found when trying to remove from the directory $directory")
+            } else {
+                throw ex
+            }
+        }
+    }
+
+    private fun isCauseNoSuchFile(ex: MessagingException): Boolean {
```

Comment:
Unfortunately, couldn't find a simpler way to determine if the error is caused by a not found file.

Reply:
Yeah, you're right to point out the difficulties you had here 👍 
If this is the best way to do it then thats fine 👍 

One minor suggestion though ... checking `innerCause.id == 2` is a bit "magic number". Nobody will know what `2` means when reading this code. Could you use the static `ChannelSftp.SSH_FX_NO_SUCH_FILE` instead please?. That is the same static that is used when `SftpException` is instantiated, so it feels like the right thing to use?